pkg/jobqueue/jobqueue.go

```diff
@@ -67,7 +67,23 @@ type JobQueue interface {
 	// finished, respectively.
 	//
 	// Lastly, the IDs of the jobs dependencies are returned.
+	// Deprecated : JobStatus, uses too much bandwidth, prefer JobStatusWoResult instead
 	JobStatus(id uuid.UUID) (jobType string, channel string, result json.RawMessage, queued, started, finished time.Time, canceled bool, deps []uuid.UUID, dependents []uuid.UUID, err error)
+	// Does the same as JobStatus but without fetching the result blob
+	// To be used instead of JobStatus if possible
+	JobStatusWoResult(id uuid.UUID) (jobType string, channel string, queued, started, finished time.Time, canceled bool, deps []uuid.UUID, dependents []uuid.UUID, err error)
+
+	// Query a specific path in a JSON field.
+	// field -> the name of the JSON field in the DB
+	// path -> a dot separated path within the field, for example osbuild_output.error
+	// response -> a pointer to the data container to Unmarshal to
+	// Returns an Unmarshalled response
+	QueryJsonField(id uuid.UUID, field string, path string, response any) error
+	// Check for presence of a specific path and is content in a JSON field.
+	// field -> the name of the JSON field in the DB
+	// path -> a dot separated path within the field, for example osbuild_output.error
+	// Returns a boolean set to True if the requested path contains data
+	TestJsonFieldExists(id uuid.UUID, field string, path string) (bool, error)
```

Comment:
I made this one specifically for OSBuildOutput, as I prefer avoiding downloading the content just to know if it exists.

Reply:
The thing is, if we rely on `ErrNoExist`  on `QueryJsonField` we will endup in the situation where we do get to download some data when we only want to have the confirmation of existence.

Reply:
Ah in the case of osbuild_output.error it makes sense yea. But I'd still exit with something predictable in case the field doesn't exist when fetching it, so we can do a nice error check.

Reply:
I'm resolving this comment because everything has changed since